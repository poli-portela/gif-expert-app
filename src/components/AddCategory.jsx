import PropTypes from "prop-types";
import React, { useState } from "react";

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  }

  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label="form">
    <input
      type="text"
      placeholder="Buscar gifs"
      value={inputValue}
      onChange={onInputChange}
    />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
