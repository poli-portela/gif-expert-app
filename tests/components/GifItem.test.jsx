import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Test de <GiftItem />", () => {
  const title = "un titulo";
  const url = "https://url.pruebas/";

  test("hacer mach con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
    
  });

  test('debe mostrar el titulo en el componente', () => { 
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
   })
});
