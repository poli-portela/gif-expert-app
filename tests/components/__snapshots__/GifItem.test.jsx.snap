// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de <GiftItem /> hacer mach con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="un titulo"
      src="https://url.pruebas/"
    />
    <p>
      un titulo
    </p>
  </div>
</div>
`;
